drivers/hmis/app/graphql/mutations/ac_hmis/calculate_alt_aha_score.rb

```diff
@@ -32,6 +31,23 @@ def resolve(enrollment_id:, form_definition_identifier:, values_by_link_id:)
         owner: enrollment,
         form_definition_identifier: form_definition_identifier,
       )
+
+      # Check for missing required responses
+      required_link_ids = aha_calculator.required_link_ids
+
+      # Use form validation to validate that the required link IDs were provided in the form submission.
+      # This lets us reuse the form validator's logic for skipping questions that weren't shown to the user
+      # (such as dependent questions whose conditions were not met).
+      definition = Hmis::Form::Definition.published.find_by(identifier: form_definition_identifier)
+      validations = definition.validate_form_values(values_by_link_id, link_ids: required_link_ids)
+
+      if validations.any?
```

Comment:
```suggestion
      if validations.reject(&:warning?).any?
```

We can ignore warning-level validations, for example items with "warn if empty"

Reply:
True thanks!